fish-rust/src/builtins/string/collect.rs

```diff
@@ -0,0 +1,71 @@
+use super::{Arguments, StringError, StringSubCommand};
+use crate::builtins::shared::{io_streams_t, STATUS_CMD_ERROR, STATUS_CMD_OK};
+use crate::ffi::{parser_t, separation_type_t};
+use crate::wchar::{wstr, L};
+use crate::wchar_ext::WExt;
+use crate::wgetopt::{wopt, woption, woption_argument_t::*};
+
+#[derive(Default)]
+pub struct Collect {
+    allow_empty: bool,
+    no_trim_newlines: bool,
+}
+
+impl StringSubCommand<'_> for Collect {
+    const LONG_OPTIONS: &'static [woption<'static>] = &[
+        wopt(L!("allow-empty"), no_argument, 'a'),
+        wopt(L!("no-trim-newlines"), no_argument, 'N'),
+    ];
+    const SHORT_OPTIONS: &'static wstr = L!(":Na");
+
+    fn parse_opt(&mut self, _optarg: Option<&wstr>, c: char) -> Result<(), StringError> {
+        match c {
+            'a' => self.allow_empty = true,
+            'N' => self.no_trim_newlines = true,
+            _ => return Err(StringError::UnknownOption),
+        }
+        Ok(())
+    }
+
+    fn handle(
+        &mut self,
+        _parser: &mut parser_t,
+        streams: &mut io_streams_t,
+        optind: &mut usize,
+        args: &[&wstr],
+    ) -> Option<libc::c_int> {
+        let mut appended = 0usize;
+
+        for (arg, want_newline) in Arguments::new(args, optind, streams, false) {
```

Comment:
nit: you dropped a `/* don't split */`

Reply:
I’d actually rather prefer to have the deciding between splitting on newline to be a different method than `new()`, like `Arguments::raw`. Both since this was a default argument in the C++, and it would make the difference more obvious than a bool. Not certain about naming, though

Reply:
maybe `without_splitting_by_newline()`